lib/pkcs11_kcv.c

```diff
@@ -31,9 +32,10 @@
 #define LABEL_WIDTH 32
 #endif
 
+
 /* target must point to a location with at least 3 bytes left */
 
-void pkcs11_display_kcv( pkcs11Context *p11Context, char *label, unsigned hmacdatasize )
+void pkcs11_display_kcv( pkcs11Context *p11Context, char *label, unsigned hmacdatasize, mac_alg_t algo, size_t num_bytes)
```

Comment:
This function is too big. Consider chunking it down into smaller support functions and doing away with the nested switch statements.

Reply:
Changes to be made in a different PR.